python/nutpie/compile_pymc.py

```diff
@@ -354,7 +354,7 @@ def expand(x, **shared):
 def compile_pymc_model(
     model: "pm.Model",
     *,
-    backend: Literal["numba", "jax"] = "numba",
+    backend: Literal["numba", "jax"] | None = None,
```

Comment:
I don't follow, why this change. The docstrings don't mention anything about `None`. The code checks for it below but it seems having numba as the explicit default is more readable?

Reply:
The code implies that None is expected. We should either have both `backend` and `gradient_backend` specified with their None or the actual default; having them being handled differently is odd.

Reply:
I think None is less bug prone because upstream defaults don't have to know what the default actually is (which can cause bugs if it ever changes).

Reply:
I'm on you with this one, but if you followed that logic to the end all arguments would have to accept None :)